lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -293,8 +293,14 @@ def get_pod_metadata(key, namespace_name, pod_name, record_create_time)
             'namespace_id' => @orphaned_namespace_id
           }
         end
-        metadata =  @cache[ids[:pod_id]] || {}
-        metadata.merge!(@namespace_cache[ids[:namespace_id]] || {}) if ids[:namespace_id]
+        metadata =  @cache.getset(ids[:pod_id]) do
+            m = fetch_pod_metadata(namespace_name, pod_name)
+            (m.nil? || m.empty?) ? nil : m
+        end || {}
+        metadata.merge!(@namespace_cache.getset(ids[:namespace_id]) do
+           m = fetch_namespace_metadata(namespace_name)
+          (m.nil? || m.empty?) ? nil : m
+        end || {})
```

Comment:
don't need the `if ids[:namespace_id]` any more?

Reply:
we will find out... rebuilt and trying to test.  i dont know what the cache does when you give it null

Reply:
simple test of passing key=nil causes the block to be invoked